src/collector-method.cpp

```diff
@@ -143,7 +143,7 @@ class Collector_Scalar : public Collector_Scalar_Base {
 
   inline void add_value(SEXP value, Path& path) {
     if (!Rf_isNull(this->transform)) value = apply_transform(value, this->transform);
-    SEXP value_casted = vec_cast(PROTECT(value), ptype);
+    SEXP value_casted = PROTECT(vec_cast(value, ptype));
```

Comment:
We still need to protect `value` because it can be allocated in `apply_transform()` above (and add +1 below).